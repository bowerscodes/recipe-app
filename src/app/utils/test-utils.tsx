import { ReactNode } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { HeroUIProvider } from "@heroui/react";
import RecipeProvider from "@/context/RecipeContext";

const AllProviders = ({ children }: { children: ReactNode }) => (
  <HeroUIProvider>
    <RecipeProvider>{children}</RecipeProvider>
  </HeroUIProvider>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
