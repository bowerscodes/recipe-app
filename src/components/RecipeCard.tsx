import { Card } from "@heroui/card";

import { Recipe } from "@/types/Recipe";
import { Image } from "@heroui/image";

interface RecipeCardProps {
  recipe: Recipe;
};

export default function RecipeCard({ recipe }: RecipeCardProps) {
  return (
    <Card className="p-4">
      <h3>{recipe.title}</h3>
      <Image 
        alt={`Image of ${recipe.title}`}
        src={recipe.image}
      />
      <div className="mt-2">
        <p>{recipe.cookingTime}</p>
        <p>{recipe.ingredients.length} Ingredients</p>
      </div>
    </Card>
  )
}

