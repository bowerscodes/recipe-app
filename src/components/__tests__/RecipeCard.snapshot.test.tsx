import { render } from "@/app/utils/test-utils";
import RecipeCard from "../RecipeCard";
import recipesData from "@/data/recipes.json";

const recipe = recipesData[0];

describe("RecipeCard Snapshots", () => {
  test("renders RecipeCard correctly", () => {
    const { container } = render(<RecipeCard recipe={recipe} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders RecipeCard with different recipe data", () => {
    const customRecipe = {
      id: 999,
      title: "Test Recipe",
      image: "https://example.com/test.jpg",
      cookingTime: "15 minutes",
      ingredients: ["Test ingredient 1", "Test ingredient 2"],
      instructions: ["Test step 1", "Test step 2"],
      reviews: [],
      author: "me"
    };

    const { container } = render(<RecipeCard recipe={customRecipe} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
