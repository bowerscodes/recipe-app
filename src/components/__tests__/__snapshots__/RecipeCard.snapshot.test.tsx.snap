// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`RecipeCard Snapshots renders RecipeCard correctly 1`] = `
<div>
  <button
    class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large cursor-pointer transition-transform-background motion-reduce:transition-none data-[pressed=true]:scale-[0.97] tap-highlight-transparent p-4"
    data-react-aria-pressable="true"
    role="button"
    tabindex="0"
    type="button"
  >
    <div
      class="flex justify-between"
    >
      <h3>
        Lasagne & Chips
      </h3>
      <span
        aria-label="Add Lasagne & Chips to favourites"
        class="p-1 cursor-pointer"
        color="black"
        role="button"
        tabindex="0"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
          />
        </svg>
      </span>
    </div>
    <div
      class="shadow-black/5 shadow-none group relative overflow-hidden bg-content3 dark:bg-content2 before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-shimmer before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 rounded-large"
      style="max-width: fit-content;"
    >
       
      <img
        alt="Image of Lasagne & Chips"
        class="relative z-10 shadow-black/5 data-[loaded=true]:opacity-100 shadow-none opacity-0 transition-transform-opacity motion-reduce:transition-none !duration-300 rounded-large"
        src="https://img2.thejournal.ie/article/5084219/river/?height=400&version=5084251"
      />
    </div>
    <div
      class="mt-2"
    >
      <p>
        45 minutes
      </p>
      <p>
        7
         Ingredients
      </p>
    </div>
  </button>
</div>
`;

exports[`RecipeCard Snapshots renders RecipeCard with different recipe data 1`] = `
<div>
  <button
    class="flex flex-col relative overflow-hidden h-auto text-foreground box-border bg-content1 outline-solid outline-transparent data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large cursor-pointer transition-transform-background motion-reduce:transition-none data-[pressed=true]:scale-[0.97] tap-highlight-transparent p-4"
    data-react-aria-pressable="true"
    role="button"
    tabindex="0"
    type="button"
  >
    <div
      class="flex justify-between"
    >
      <h3>
        Test Recipe
      </h3>
      <span
        aria-label="Add Test Recipe to favourites"
        class="p-1 cursor-pointer"
        color="black"
        role="button"
        tabindex="0"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
          />
        </svg>
      </span>
    </div>
    <div
      class="shadow-black/5 shadow-none group relative overflow-hidden bg-content3 dark:bg-content2 before:opacity-100 before:absolute before:inset-0 before:-translate-x-full before:animate-shimmer before:border-t before:border-content4/30 before:bg-gradient-to-r before:from-transparent before:via-content4 dark:before:via-default-700/10 before:to-transparent after:opacity-100 after:absolute after:inset-0 after:-z-10 after:bg-content3 dark:after:bg-content2 rounded-large"
      style="max-width: fit-content;"
    >
       
      <img
        alt="Image of Test Recipe"
        class="relative z-10 shadow-black/5 data-[loaded=true]:opacity-100 shadow-none opacity-0 transition-transform-opacity motion-reduce:transition-none !duration-300 rounded-large"
        src="https://example.com/test.jpg"
      />
    </div>
    <div
      class="mt-2"
    >
      <p>
        15 minutes
      </p>
      <p>
        2
         Ingredients
      </p>
    </div>
  </button>
</div>
`;
